src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -1061,11 +1062,20 @@ public void OnSearchElementClicked(NodeModel nodeModel, Point position)
 
         internal void OnRequestConnectToPort(string nodeCreationName, PortModel portModel)
         {
+            // Do not auto connect code block node since default code block node do not have output port
             if (!nodeCreationName.Contains("Code Block"))
             {
+                //dynamoViewModel.ExecuteCommand(new DynamoModel.MakeConnectionCommand(nodeModel.GUID,))
+                var id = Guid.NewGuid();
                 // Create a new node based on node creation name and connect ports
-                dynamoViewModel.ExecuteCommand(new DynamoModel.CreateAndConnectNodeCommand(Guid.NewGuid(), portModel.Owner.GUID,
-                    nodeCreationName, 0, portModel.Index, portModel.CenterX - 50, portModel.CenterY + 200, false, false));
+                var initialNode = dynamoViewModel.CurrentSpaceViewModel.Nodes.Where(x => x.Id == portModel.Owner.GUID).FirstOrDefault();
```

Comment:
isn't `initialNode` the same as `portModel.Owner`?

Reply:
Can we split this step into a `CreateNodeCommand` step and then add a new command called something like `PlaceAndConnectCommand`? That way we can first create the new node, get its height and width and place it relative to the original node according to its respective height and width. In other words the `PlaceAndConnectCommand` will be a combination of 2 steps: 1) place the node based on an calculated X and Y position and 2) call the `MakeConnectionCommand`.

Reply:
This way we might be able to do away with these hard-coded values.